### PR TITLE
EXP-985 Exposed Token Transfer Utilities 

### PR DESCRIPTION
**Pull request by ahsankhan201010**

### CODE REVIEW

**Comment by NidaKarimAli** [comment]
@ahsankhan201010 please also bump the version. Also the build is failing, do we need a library version bump?

**Comment by ahsankhan201010** [comment]
@NidaKarimAli bumped the version for client but we need to bump the version of library-sui as well before merging. The PR for library-sui is still in review, will push the bump commit for library-sui once its PR got merged :rocket:

File: src/bluefinClient.ts [hunk 1]
```diff
@@ -859,6 +859,17 @@ export class BluefinClient {
     );
   };
 
+  /**
+   * @param walletAddress wallet address of the user
+   * @returns string
+   * @description
```
Comment (YameenMalik) [inline src/bluefinClient.ts:865]: description goes at the top of the comment

Comment (YameenMalik) [reply src/bluefinClient.ts:865]: I see in all the newly added methods the `description` comment is at the bottom. Please remedy it. Thanks

File: src/bluefinClient.ts [hunk 2]
```diff
@@ -2008,4 +2019,60 @@ export class BluefinClient {
...
   };
+
+  /**
+   * @param to recipient wallet address
+   * @param balance SUI amount to transfer
+   * @returns Response Schema
+   * @description
+   * transfer sui tokens
+   * */
+  transferSuiBalance = async (
```
Comment (YameenMalik) [inline src/bluefinClient.ts:2030]: Please make a single method `transferCoins` the method can expect the name of the coin as input and perform switch on the name

File: src/bluefinClient.ts [hunk 3]
```diff
@@ -2008,4 +2019,60 @@ export class BluefinClient {
...
+  };
+
+  /**
+   * @param to recipient wallet address
+   * @param balance amount to transfer
+   * @returns BigInt
+   * @description
+   * estimate gas for sui token transfer
+   * */
+  estimateGasForSuiTransfer = async (
```
Comment (YameenMalik) [inline src/bluefinClient.ts:2058]: please make a single method for estimation of gas that should take a transaction block as input, dry runs it and returns the estimated gas fee amount.

**Review by YameenMalik — CHANGES_REQUESTED** [review]
Please entertain comments before merger. thanks

File: src/bluefinClient.ts (hunk 2 above)
Comment (ahsankhan201010) [reply src/bluefinClient.ts]: done :rocket:

File: src/bluefinClient.ts (hunk 1 above)
Comment (ahsankhan201010) [reply src/bluefinClient.ts:864]: done :rocket:

File: src/bluefinClient.ts (hunk 3 above)
Comment (ahsankhan201010) [reply src/bluefinClient.ts:2046]: I already declared a utility estimateGas in library sui that takes tx as an argument and returns the estimated gas. But I exposed two separate utilities for estimation of gas for sui and usdc transfer.

If I need to expose estimate gas utility then I also need to expose prepareTransactionForUSDCTransfer & prepareTransactionForSUITransfer and call them on FE when input change and pass it to estimateGas utility. I that case we have to expose three utilities

- estimateGas
- prepareTransactionForUSDCTransfer
- prepareTransactionForSUITransfer

That's why decided to expose estimateGasForUSDCTransfer & estimateGasForSUITransfer which is calling estimateGas and prepareTrasansaction internally.

**Review by YameenMalik — APPROVED** [review]

**Review by NidaKarimAli — APPROVED** [review]

**Pull request merged**